dashboard/src/components/TooltipDateTime/TooltipDateTime.tsx

```diff
@@ -0,0 +1,64 @@
+import { format } from 'date-fns';
+
+import { Tooltip, TooltipContent, TooltipTrigger } from '@/components/Tooltip';
+import { getDateOffset } from '@/utils/utils';
+
+type TooltipDateTimeProps = {
+  dateTime: string;
+  forceNoDate?: boolean;
+  dateFormat?: string;
+  timeFormat?: string;
+  lineBreak?: boolean;
+  showTime?: boolean;
+  showTZ?: boolean;
+  showTooltip?: boolean;
+};
+
+const TooltipDateTime = ({
+  dateTime,
+  forceNoDate,
+  dateFormat,
+  timeFormat,
+  lineBreak,
+  showTime,
+  showTZ,
+  showTooltip,
+}: TooltipDateTimeProps): JSX.Element => {
+  const dateObj = new Date(dateTime);
+
+  let date = dateFormat
+    ? format(dateObj, dateFormat)
+    : dateObj.toLocaleDateString();
+  let time = timeFormat
+    ? format(dateObj, timeFormat)
+    : dateObj.toLocaleTimeString();
+  let tz = getDateOffset(dateObj);
+
+  if (forceNoDate) {
+    date = '-';
+    time = tz = '';
+  }
+
+  return (
+    <Tooltip>
+      <TooltipTrigger>
+        <div>
+          {date} {showTime ? time : ''} {showTZ ? tz : ''}
+        </div>
+      </TooltipTrigger>
+      {showTooltip === false ? (
+        ''
+      ) : (
+        <TooltipContent>
+          <div className="text-center">
+            {date}
+            {lineBreak ? <br /> : ' '}
+            {time} {timeFormat && dateFormat ? '' : tz}
+          </div>
+        </TooltipContent>
+      )}
```

Comment:
In react it is common to use short circuit with the `and` operator (`&&`), it is also possible to use with `or` but that is not common

`showTootltip && (component)` 

https://react.dev/learn/conditional-rendering